File: app/src/main/java/com/qyt/accumulation/ui/activity/StatisticsActivity.java
package com.qyt.accumulation.ui.activity;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.widget.Toolbar;

import com.qyt.accumulation.R;
import com.qyt.accumulation.base.BaseActivity;
import com.qyt.accumulation.entity.Goal;
import com.qyt.accumulation.ui.fragment.ChartFragment;
import com.qyt.accumulation.util.Temp;

import java.util.ArrayList;
import java.util.List;

/**
 * 统计
 * Created by Relish on 2016/11/10.
 */
public class StatisticsActivity extends BaseActivity implements ViewPager.OnPageChangeListener {
    @Override
    protected int layoutId() {
        return R.layout.activity_statistics;
    }

    @Override
    protected void initToolbar(Bundle savedInstanceState, Toolbar mToolbar) {

    }

    private ViewPager vp;
    private ArrayList<Fragment> mFragments;
    private List<Goal> mGoals;

    @Override
    protected void initViews(Bundle savedInstanceState) {
        vp = (ViewPager) findViewById(R.id.vp);
        mFragments = new ArrayList<>();
        mGoals = Goal.findAll();
        for (Goal goal : mGoals) {
            mFragments.add(ChartFragment.getInstance(goal));
        }
        FragmentsAdapter adapter = new FragmentsAdapter(getSupportFragmentManager());
        vp.setAdapter(adapter);
        vp.setOnPageChangeListener(this);
        vp.setOffscreenPageLimit(Temp.getGoals().size());//保留n个界面
        selectPage(0);
    }


    private void selectPage(int index) {
        vp.setCurrentItem(index);
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

    }

    @Override
    public void onPageSelected(int position) {
        selectPage(position);
    }

    @Override
    public void onPageScrollStateChanged(int state) {

    }

    /**
     * pageAdapter
     */
    class FragmentsAdapter extends FragmentPagerAdapter {

        public FragmentsAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public android.support.v4.app.Fragment getItem(int position) {
            return mFragments.get(position);
        }

        @Override
        public int getCount() {
            return mFragments.size();
        }
    }
}

File: app/src/main/java/com/qyt/accumulation/ui/activity/StatisticsActivity.kt
package com.qyt.accumulation.ui.activity

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.view.ViewPager
import android.support.v7.widget.Toolbar
import android.view.View
import android.widget.TextView

import com.qyt.accumulation.R
import com.qyt.accumulation.base.BaseActivity
import com.qyt.accumulation.entity.Goal
import com.qyt.accumulation.ui.fragment.ChartFragment

import java.util.ArrayList

/**
 * 统计
 * Created by Relish on 2016/11/10.
 */
class StatisticsActivity : BaseActivity(), ViewPager.OnPageChangeListener {
    override fun layoutId(): Int {
        return R.layout.activity_statistics
    }

    override fun initToolbar(savedInstanceState: Bundle?, mToolbar: Toolbar?) {
        mToolbar?.title = "统计"
    }

    private var vp: ViewPager? = null
    private var mFragments: ArrayList<Fragment>? = null
    private var mGoals: List<Goal>? = null
    private var tv_no_data: TextView? = null

    override fun initViews(savedInstanceState: Bundle?) {
        tv_no_data = findViewById(R.id.tv_no_data) as TextView
        vp = findViewById(R.id.vp) as ViewPager
        mFragments = ArrayList<Fragment>()
        mGoals = Goal.findAll()
        for (goal in mGoals!!) {
            mFragments!!.add(ChartFragment.getInstance(goal))
        }
        val adapter = FragmentsAdapter(supportFragmentManager)
        vp!!.adapter = adapter
        vp!!.setOnPageChangeListener(this)
        vp!!.offscreenPageLimit = mGoals!!.size//保留n个界面
        selectPage(0)
    }

    override fun onResume() {
        super.onResume()
        if (mGoals == null || mGoals!!.size == 0) {
            tv_no_data!!.visibility = View.VISIBLE
            vp!!.visibility = View.GONE
        } else {
            tv_no_data!!.visibility = View.GONE
            vp!!.visibility = View.VISIBLE
        }
    }

    private fun selectPage(index: Int) {
        vp!!.currentItem = index
    }

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

    }

    override fun onPageSelected(position: Int) {
        selectPage(position)
    }

    override fun onPageScrollStateChanged(state: Int) {

    }

    /**
     * pageAdapter
     */
    internal inner class FragmentsAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

        override fun getItem(position: Int): android.support.v4.app.Fragment {
            return mFragments!![position]
        }

        override fun getCount(): Int {
            return mFragments!!.size
        }
    }
}
